sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -1182,6 +1188,11 @@ def run_Read(self, transform_node, options):
         raise ValueError(
             'BigQuery source is not currently available for use '
             'in streaming pipelines.')
+      debug_options = options.view_as(DebugOptions)
+      use_fn_api = (debug_options.experiments and
+                    'beam_fn_api' in debug_options.experiments)
+      if use_fn_api:
```

Comment:
Should we also check for experiment use_runner_v2 or is this adequate ?

Reply:
Might make sense to, perhaps even only check use_runner_v2, since that's the top-level option we're expecting users to provide (and then we internally add beam_fn_api).

Reply:
This is enough. For beam_fn_api is added along with use_runner_v2.

Reply:
oops Harsh , I did not see your message. ^^'

But the source is not supported on FnApi either, though, right? so that would also help users trying Batch FnApi jobs?

Reply:
In general, yes, but the only support way of batch fnapi is with UW -- they shouldn't be trying fnapi batch jobs without it. That said, perhaps it makes sense to check both separately (just to ensure that the --use_runner_v2 also independently triggers this check, just in case someone in the future decides to accidentally modify something)?

Reply:
The unit test that I've added checks both of them separately, so if either of them changes, the unit test will fail. WDYT about that?

Reply:
Yes, that sounds great, thanks!

Reply:
One thing I want to ensure (since I'm not completely familiar with the error semantics of the functions in Python), is that this error will stop the pipeline from being submitted, and essentially be the equivalent of a C++ CheckFail? We want this error to stop everything, otherwise users might miss this.

Reply:
Yes Harsh, this will stop the pipeline from being submitted, and fail the command. The unit test also goes through this flow (trying to submit to Dataflow, and having the pipeline throw an error, and not be submitted).